feature/doc-59

shikhanansi opened the pull request:
Hi Morgan,
Please review the new chapter and provide feedback. 
Kind Regards,
Shikha

ephox-mogran commented:
Hey,

Some subjective feedback :)

A few notes:

1. The annotator API isn't a plugin, so we probably shouldn't call it that.
2. The "comment" and the "author" examples are being used interchangeably
3. Code examples have inconsistent indentation

A few minor wording issues:

- "The TinyMCE Annotation plugin provides flexibility in the appearance and placement of your comments" isn't really relevant.
- use retrieve instead of retrieving in first paragraph
- multiple annotation formats ... formats isn't quite the right word
- uid *may* require an explanation
- Section 3's text needs a bit of work
- Change "full annotate API" to something about registration, as it isn't the full API

shikhanansi commented:
Hi Morgan,
I have updated the doc to accommodate your feedback. Please review the updates. I have also added the note about user warning that is generated because of using "experimental".
Regards,
Shikha

tinydylan commented on enterprise/tiny-comments.md:
```diff
@@ -0,0 +1,30 @@
+layout: default
+title: Tiny Comments
+description: Tiny Comments provides the ability to add comments to the content and collaborate with other users for content editing.
+keywords: enterprise pricing video youtube vimeo mp3 mp4 mov movie clip film comment commenting mediaembed media
```
The video-related keywords are not relevant to this page.

ephox-dayne replied:
"collaborate with other users for content editing" doesn't read correctly (to me). 
- Tiny Comments allows content editors to collaborate by providing the ability to add comments to the content.
- Tiny Comments allows users to collaborate by providing the ability to add comments to the content.
- Tiny Comments provides the ability to add comments to content being edited in TinyMCE. This allows TinyMCE users to collaborate without ever leaving the editor.

tinydylan commented on enterprise/tiny-comments.md:
```diff
@@ -0,0 +1,30 @@
+keywords: enterprise pricing video youtube vimeo mp3 mp4 mov movie clip film comment commenting mediaembed media
+---
+
+The [Tiny Comment]({{ site.baseurl }}/plugins/tiny-comment/) plugin provides the ability to add comments to the content and collaborate with other users for content editing within the TinyMCE editor.
```
Plugin is named "Tiny Comments".
URL is tiny-comments

tinydylan commented on enterprise/tiny-comments.md:
```diff
@@ -0,0 +1,30 @@
+---
+
+The [Tiny Comment]({{ site.baseurl }}/plugins/tiny-comment/) plugin provides the ability to add comments to the content and collaborate with other users for content editing within the TinyMCE editor.
+The primary value that TinyComments offer is a user interface to create, reply, and delete comments. The TinyComments plugin is built upon the the new [Annotations API]({{ site.baseurl }}/plugins/annotations/) and uses annotations to create comment threads.
```
"Tiny Comments", not "TinyComments"

tinydylan replied:
"create, reply, and delete"

We create comments.
We delete comments.
We reply *to* comments.

Suggested wording:
"create, delete and reply to comments".

tinydylan commented on enterprise/tiny-comments.md:
```diff
@@ -0,0 +1,30 @@
+       selector: 'textarea',
+       toolbar: 'bold italic underline | tinycomments',
+       plugins: 'tinycomments',
+       content_style: '.mce-annotation { background: yellow; color: black; } .tc-active-annotation {background: lime; color: black; }',
```
@ephox-mogran This API looks strange to me. Why are all the "commenting" capabilities labelled as "annotations"? What if the user has a different annotation handler? 

I would have expected the commenting plugin to consume the "annotations" api, and produce a "commenting" api.

tinydylan commented on enterprise/tiny-comments.md:
```diff
@@ -0,0 +1,30 @@
+       annotations_create: (author, content, done, fail) => { console.log("comment created"); done(); },
+       annotations_reply: (uid, author, content, done, fail) => { console.log("comment reply"); done(); },
+       annotations_get: (done, fail) => { console.log("comment get"); done(); },
+       annotations_delete: (uid, done, fail) => { console.log("comment del"); done(); },
```
Should we be putting code examples in ES3?

tinydylan commented on plugins/annotations.md:
```diff
@@ -0,0 +1,226 @@
+---
```
Per Shikha's comments, annotations is to be reviewed in #728

ephox-dayne commented on enterprise/tiny-comments.md:
```diff
@@ -0,0 +1,75 @@
+---
+
+The [Tiny Comment]({{ site.baseurl }}/plugins/tiny-comment/) plugin provides the ability to add comments to the content and collaborate with other users for content editing within the TinyMCE editor.
+The primary value that TinyComments offer is a user interface to create, reply, and delete comments. The TinyComments plugin is built upon the the new [Annotations API]({{ site.baseurl }}/plugins/annotations/) and uses annotations to create comment threads.
```
- "The primary value"? What about just "Tiny Comments offers a user interface.."
- If you keep it the way it is, it should be "The primary value that Tiny Comments offer**s**"

ephox-dayne commented on enterprise/tiny-comments.md:
```diff
@@ -0,0 +1,75 @@
+The [Tiny Comment]({{ site.baseurl }}/plugins/tiny-comment/) plugin provides the ability to add comments to the content and collaborate with other users for content editing within the TinyMCE editor.
+The primary value that TinyComments offer is a user interface to create, reply, and delete comments. The TinyComments plugin is built upon the the new [Annotations API]({{ site.baseurl }}/plugins/annotations/) and uses annotations to create comment threads.
+
+> Note: The TinyComments plugin in its current state does not store the comments anywhere. However, it gives the integrator callback APIs to create, search, and store comment threads.
```
Need to decide if Tiny Comments is one word or two and write it consistently.

ephox-dayne commented on enterprise/tiny-comments.md:
```diff
@@ -0,0 +1,75 @@
+
+> Note: The TinyComments plugin in its current state does not store the comments anywhere. However, it gives the integrator callback APIs to create, search, and store comment threads.
+
+Example:
```
Worth noting that this example will just store the comments in memory. If the user refreshes the page in the browser then the comments will disappear.

Also, this example won't run as is, it's not a complete HTML document and doesn't include the editor. Do we want examples that readers copy and paste and go?

ephox-dayne commented on enterprise/tiny-comments.md:
```diff
@@ -0,0 +1,75 @@
+            selector: "#textarea",
+            toolbar: 'bold italic underline | tinycomments',
+            plugins: "tinycomments",
+            content_style: '.mce-annotation { background: yellow; color: black; } .tc-active-annotation {background: lime; color: black; }',
```
This is required so that commented on text is highlighted in the editor. That may not be immediately obvious to a reader (it wasn't to me).

ephox-dayne commented:
Other plugins have a section for each tinymce.init option. Is that how we want this to be? See https://www.tiny.cloud/docs/plugins/linkchecker/
